includes/frontend.php

```diff
@@ -1163,11 +1163,32 @@ protected function get_init_settings() {
 
 			$title = Utils::urlencode_html_entities( wp_get_document_title() );
 
+			$thumb_id = get_post_thumbnail_id();
+			$thumb_size = 'full';
+			$full_thumb_meta = wp_get_attachment_image_src( $thumb_id, $thumb_size );
+
+			// If thumb is wider than 1600px, get the closest thumbnail size to it
+			// This is done because Pinterest has a problem sharing large images
+			if ( $thumb_id && $full_thumb_meta && ( 1600 < $full_thumb_meta[1] ) ) {
+				$thumbnail_sizes = Group_Control_Image_Size::get_all_image_sizes();
+
+				uasort( $thumbnail_sizes, function( $a, $b ) {
+					return $b['width'] - $a['width'];
+				} );
+
+				foreach ( $thumbnail_sizes as $size_slug => $size_meta ) {
+					if ( 1600 > $size_meta['width'] ) {
+						$thumb_size = $size_slug;
+						break;
+					}
+				}
+			}
+
 			$settings['post'] = [
 				'id' => $post->ID,
 				'title' => $title,
 				'excerpt' => $post->post_excerpt,
-				'featuredImage' => get_the_post_thumbnail_url(),
+				'featuredImage' => get_the_post_thumbnail_url( null, $thumb_size ),
```

Comment:
It should be used only for Pinterest, add it as an additional setting. and extract the logic to a separated method.

Reply:
I think I will just try to pull the 'large' size thumbnail, and have the 'full' as a fallback - this is more economical, and less logic